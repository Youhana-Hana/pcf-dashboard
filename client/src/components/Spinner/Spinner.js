import React, { Component } from 'react';
import { HashLoader } from 'react-spinners';

class Spinner extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
        <div className='sweet-loading'>
        <HashLoader
      color={'#000000'}
      size= { parseInt(this.props.size)}
      loading={Boolean(this.props.loading)} 
        />
        </div>
    );
  }
}

export default Spinner;
