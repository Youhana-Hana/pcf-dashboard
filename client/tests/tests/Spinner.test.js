import React from 'react';
import renderer from 'react-test-renderer';
import Spinner  from '../../src/components/Spinner/';

it('test to see if the spinner renders correctly', () => {
  const spinner = renderer
        .create(<Spinner size="12" loading="true" />)
  .toJSON();
  expect(spinner).toMatchSnapshot();
});
