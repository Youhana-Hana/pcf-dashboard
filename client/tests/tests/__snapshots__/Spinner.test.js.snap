// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the spinner renders correctly 1`] = `
<div
  className="sweet-loading col-centered"
>
  <div
    className="css-fmrsd2"
  >
    <div
      className="css-1h0xra1"
    />
    <div
      className="css-s5vm1e"
    />
  </div>
</div>
`;
